common/src/main/java/com/sedmelluq/lava/common/natives/architecture/DefaultOperatingSystemTypes.java

```diff
@@ -34,16 +45,43 @@ public String libraryFileSuffix() {
   public static OperatingSystemType detect() {
     String osFullName = System.getProperty("os.name");
 
-    if (osFullName.startsWith("Windows")) {
+    if(osFullName.startsWith("Windows")) {
       return WINDOWS;
-    } else if (osFullName.startsWith("Mac OS X")) {
+    } else if(osFullName.startsWith("Mac OS X")) {
       return DARWIN;
-    } else if (osFullName.startsWith("Solaris")) {
+    } else if(osFullName.startsWith("Solaris")) {
       return SOLARIS;
-    } else if (osFullName.toLowerCase().startsWith("linux")) {
-      return LINUX;
+    } else if(osFullName.toLowerCase().startsWith("linux")) {
+      return checkMusl() ? LINUX_MUSL : LINUX;
     } else {
       throw new IllegalArgumentException("Unknown operating system: " + osFullName);
     }
   }
+
+  private static boolean checkMusl() {
+    Boolean b = cachedMusl;
+    if(b == null) {
+      synchronized(DefaultOperatingSystemTypes.class) {
+        boolean check;
+        try {
+          Process p = new ProcessBuilder("ldd", "--version")
+                  .start();
+          log.debug("Exit code: {}", p.waitFor());
+          String line = new BufferedReader(new InputStreamReader(p.getInputStream())).readLine();
+          log.debug("First line of stdout: {}", line);
+          if(line == null) {
+            line = new BufferedReader(new InputStreamReader(p.getErrorStream())).readLine();
+            log.debug("First line of stderr: {}", line);
+          }
+          check = line != null && line.toLowerCase().startsWith("musl");
+        } catch(IOException | InterruptedException fail) {
```

Comment:
```suggestion
        try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/self/maps")))) {
          String line;
          while ((line = reader.readLine()) != null) {
            if (line.contains("-musl-")) {
              check = true;
            }
          }
        } catch(IOException fail) {
```

How about this solution instead of `java.lang.Process`?

Reply:
Interestingly this exactly fixes an issue I ran into. See [here](https://xgob.in/edcy1jad)